pkg/forwarder/forwarder.go

```diff
@@ -346,3 +383,87 @@ func (f *DefaultForwarder) SubmitV1Intake(payload Payloads, extra http.Header) e
 	transactionsIntakeV1.Add(1)
 	return f.sendHTTPTransactions(transactions)
 }
+
+// SubmitProcessChecks sends process checks
+func (f *DefaultForwarder) SubmitProcessChecks(payload Payloads, extra http.Header) (chan Response, error) {
+	transactionsIntakeProcesses.Add(1)
+
+	return f.submitProcessLikePayload(processesEndpoint, payload, extra, true)
+}
+
+// SubmitRTProcessChecks sends real time process checks
+func (f *DefaultForwarder) SubmitRTProcessChecks(payload Payloads, extra http.Header) (chan Response, error) {
+	transactionsIntakeRTProcesses.Add(1)
+
+	return f.submitProcessLikePayload(rtProcessesEndpoint, payload, extra, false)
+}
+
+// SubmitContainerChecks sends container checks
+func (f *DefaultForwarder) SubmitContainerChecks(payload Payloads, extra http.Header) (chan Response, error) {
+	transactionsIntakeContainer.Add(1)
+
+	return f.submitProcessLikePayload(containerEndpoint, payload, extra, true)
+}
+
+// SubmitRTContainerChecks sends real time container checks
+func (f *DefaultForwarder) SubmitRTContainerChecks(payload Payloads, extra http.Header) (chan Response, error) {
+	transactionsIntakeRTContainer.Add(1)
+
+	return f.submitProcessLikePayload(rtContainerEndpoint, payload, extra, false)
+}
+
+// SubmitConnectionChecks sends connection checks
+func (f *DefaultForwarder) SubmitConnectionChecks(payload Payloads, extra http.Header) (chan Response, error) {
+	transactionsIntakeConnections.Add(1)
+
+	return f.submitProcessLikePayload(connectionsEndpoint, payload, extra, true)
+}
+
+// SubmitPodChecks sends pod checks
+func (f *DefaultForwarder) SubmitPodChecks(payload Payloads, extra http.Header) (chan Response, error) {
+	transactionsIntakePod.Add(1)
+
+	return f.submitProcessLikePayload(podEndpoint, payload, extra, true)
+}
+
+func (f *DefaultForwarder) submitProcessLikePayload(ep endpoint, payload Payloads, extra http.Header, retryable bool) (chan Response, error) {
+	transactions := f.createHTTPTransactions(ep, payload, false, extra)
+
+	results := make(chan Response, len(transactions))
+
+	var wg sync.WaitGroup
```

Comment:
What's async here that a wg is needed?

Reply:
Submitting the "http transaction" to the workers.  We will create txn per (domain,  apiKey) combination and each will be submitted. separately.

When all. the requests complete, the results channel will be closed (via a separate goroutine waiting on the wait group)

Reply:
Ah got it.